### ISSUE TITLE
Tearsheet: support aria-label

### ISSUE THREAD

**Issue by wkeese** [carbon-design-system/ibm-products#4013]
## What package(s) are you using?

- [x] Carbon for IBM Products (or Carbon for IBM Cloud & Cognitive)
- [ ] Legacy/CDAI
- [ ] Legacy/Security

## Detailed description

> Describe in detail the issue you're having.

Even after the fix to #2978, my tearsheet is getting an invalid `aria-label`.  A simplified version of my code would be

```tsx
<Tearsheet title={<div><span>hello</span> <span>world</span> <Button>click me</Button></div>} ...>
```

which ends up generating something like

```html
<div class="cds--modal-container cds--modal-container--sm c4p--tearsheet__container c4p--tearsheet__container--lower"
role="dialog" aria-modal="true" aria-label="helloworldclick me">
```

In my actual code, the spaces between the elements gets erased, although that doesn't seem to happen with simpler examples.

(Note: Admittedly I should be using `headerActions`, but I think supporting an explicit `aria-label` still makes sense.)

> Is this issue related to a specific component?

Yes, Tearsheet.

> What did you expect to happen? What happened instead? What would you like to
> see changed?

I expected `aria-label="hello world click me"` (with spaces), but what I really want is support for an explicit `aria-label` property.  The reason is that my actual `title` property contains a button that is conceptually not part of the Tearsheet's title.

> What browser are you working in?

Chrome

> What version of the @carbon/ibm-products (or @carbon/ibm-cloud-cognitive)
> package are you using?

2.15.1

> What offering/product do you work on? Any pressing ship or release dates we
> should be aware of?

IBM Knowledge Catalog

## Steps to reproduce the issue

Try the sample below and inspect using devtools.

> Please create a reduced test case in CodeSandbox

https://codesandbox.io/p/devbox/wild-sky-h7jm6x

```[tasklist]
### Acceptance criteria
- [x] If not `aria-label` is set, use `title`
- [x] If `aria-label` is passed, use `aria-label`
```
